wezterm-gui/src/termwindow/mod.rs

```diff
@@ -2507,28 +2507,33 @@ impl TermWindow {
             ToggleFullScreen => {
                 self.window.as_ref().unwrap().toggle_fullscreen();
             }
-            #[cfg(target_os = "macos")]
-            ToggleFloatingWindow => {
+            SetWindowLevel(level) => {
                 let window = self.window.clone().unwrap();
+                let is_always_on_top = self.window_state.intersects(WindowState::ALWAYS_ON_TOP);
+                let is_always_on_bottom = self.window_state.intersects(WindowState::ALWAYS_ON_BOTTOM);
```

Comment:
For this, what I'd envisioned is adding a method to `WindowState` that returns the effective `WindowLevel`:

```rust
impl WindowState {
   pub fn as_window_level(self) -> WindowLevel {
      if self.contains(Self::ALWAYS_ON_TOP) {
         WindowLevel::AlwaysOnTop
      } else if self.contains(Self::ALWAYS_ON_BOTTOM) {
         WindowLevel::AlwaysOnBotttom
      } else {
         WindowLevel::Normal
      }
   }
}
```

then code over here can much more simply reason about the current level